src/dialogs/enter-code/dialog-enter-code.ts

```diff
@@ -52,14 +52,18 @@ export class DialogEnterCode
   }
 
   private _submit(): void {
-    if (
-      this._dialogParams &&
-      this._dialogParams.codePattern &&
-      RegExp(this._dialogParams.codePattern).test(this._input?.value ?? "")
-    ) {
-      this._dialogParams?.submit?.(this._input?.value ?? "");
-      this.closeDialog();
+    if (this._dialogParams && this._dialogParams.codePattern) {
+      if (
+        RegExp(this._dialogParams.codePattern).test(this._input?.value ?? "")
+      ) {
+        this._dialogParams?.submit?.(this._input?.value ?? "");
+        this.closeDialog();
+      } else {
+        this.closeDialog();
```

Comment:
I don't think we want to close the dialog in this case, but show an error to the user?